src/dispatch/case/flows.py

```diff
@@ -384,3 +384,41 @@ def case_to_incident_escalate_flow(
         description=f"The members of the incident's tactical group {incident.tactical_group.email} have been given permission to access the case's storage folder",
         case_id=case.id,
     )
+
+
+def case_escalate_flow(case_id: PrimaryKey, incident_id: PrimaryKey, db_session: SessionLocal):
+    """Allows for a case to be escalated to an incident while modifying it's properties."""
+    case = get(case_id=case_id, db_session=db_session)
+    incident = incident_flows.incident_create_flow(incident_id=incident_id, db_session=db_session)
+
+    case.escalated_at = datetime.utcnow()
+    case.status = CaseStatus.escalated
+    case.incidents.append(incident)
+
+    db_session.add(case)
+    db_session.commit()
+
+    event_service.log_case_event(
+        db_session=db_session,
+        source="Dispatch Core App",
+        description=f"The case has been linked to incident {incident.name} in the {incident.project.name} project",
+        case_id=case.id,
+    )
+
+    # we add the incident's tactical group to the case's storage folder
+    # to allow incident participants to access the case's artifacts in the folder
+    storage_members = [incident.tactical_group.email]
+    storage_flows.update_storage(
+        obj=case,
+        storage_action=StorageAction.add_members,
+        storage_members=storage_members,
+        db_session=db_session,
+    )
+
+    event_service.log_case_event(
+        db_session=db_session,
+        source="Dispatch Core App",
+        description=f"The members of the incident's tactical group {incident.tactical_group.email} have been given permission to access the case's storage folder",
+        case_id=case.id,
+    )
```

Comment:
This is copypasta from the `case_to_incident_escalate_flow` function. Should we create a ` case_to_incident_escalate_flow_common` function and move this logic there and call the function from both `case_to_incident_escalate_flow` and `case_to_incident_escalate_custom_flow`?